fendermint/app/src/app.rs

```diff
@@ -803,22 +813,31 @@ where
         let validator_updates =
             to_validator_updates(power_updates.0).context("failed to convert validator updates")?;
 
-        // If the block gas limit has changed, we need to update the consensus layer so it can
-        // pack subsequent blocks against the new limit.
-        let consensus_param_updates = {
-            let mut consensus_params = self
-                .client
-                .consensus_params(tendermint::block::Height::try_from(request.height)?)
-                .await?
-                .consensus_params;
-
-            if consensus_params.block.max_gas != new_block_gas_limit as i64 {
-                consensus_params.block.max_gas = new_block_gas_limit as i64;
-                Some(consensus_params)
+        // Updates the validator cache with the new validator set.
+        let validators_ids_with_keys: Vec<(TendermintAccountId, TendermintPublicKey)> =
+            validator_updates
```

Comment:
If i rmb correctly, `validator_updates` is the power change, so it's about changes, not setting validators.

Reply:
Hmm. In that case where would be a good place to get the validator changes?

Reply:
So we just need to check if the power is 0, that means the validator is removed, we should purge instead. See: https://github.com/consensus-shipyard/ipc/blob/409c1387ddf978a76c999bd8a19c7291dd094fd1/fendermint/vm/interpreter/src/fvm/checkpoint.rs#L434. Else it's just an `upsert`.